assets/src/edit-story/components/dropTargets/withDropTarget.js

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useDrop } from 'react-dnd';
+
+/**
+ * Internal dependencies
+ */
+import { useEffect, useCallback, useState } from 'react';
+import StoryPropTypes from '../../types';
+import { MASKS } from '../../masks/index';
+import { useStory } from '../../app/story';
+
+const dropTargetStyles = {
+  transition: 'opacity 0.5s',
+  pointerEvents: 'visibleStroke',
+};
+
+const dropTargetContainerStyle = {
+  position: 'absolute',
+  top: 0,
+  left: 0,
+  width: '100%',
+  height: '100%',
+  pointerEvents: 'none',
+};
+
+function WithDropTarget({ element, children }) {
+  const {
+    actions: { updateElementById },
+  } = useStory();
+  const setProperties = useCallback(
+    (properties) => updateElementById({ elementId: element.id, properties }),
+    [element.id, updateElementById]
+  );
+
+  const [initialBackgroundImage, setInitialBackgroundImage] = useState(
+    element.backgroundImage
+  );
+
+  const [{ isOver, item }, drop] = useDrop({
+    accept: 'media',
+    drop: () => {
+      setInitialBackgroundImage(item.id);
+      setProperties({
+        backgroundImage: item.id,
+      });
+    },
+    collect: (monitor) => ({
+      isOver: monitor.isOver({ shallow: true }),
+      item: monitor.getItem(),
+    }),
+  });
+
+  // Previews
+  useEffect(() => {
+    if (!isOver || !item) {
+      // Reset background image when the element is out of the drop target
+      if (element.backgroundImage !== initialBackgroundImage) {
+        setProperties({ backgroundImage: initialBackgroundImage });
+      }
+      return;
+    }
+    // Preview the background image
+    setProperties({
+      backgroundImage: item.id,
+    });
+  }, [
+    item,
+    isOver,
+    setProperties,
+    element.backgroundImage,
+    initialBackgroundImage,
+  ]);
+
+  const mask = element.mask;
+
+  if (mask && mask.type && MASKS[mask.type]) {
+    return (
+      <>
+        {children}
+        <svg
+          style={dropTargetContainerStyle}
+          viewBox="0 0 1 1"
+          width="100%"
+          height="100%"
+          preserveAspectRatio="none"
+        >
+          <path
+            ref={drop}
+            vectorEffect="non-scaling-stroke"
+            strokeWidth="32"
```

Comment:
The wide stroke goes on both sides: outside and inside, right? Is that always a good effect?